daemon/state.go

```diff
@@ -259,118 +254,10 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 
 	for _, ep := range state.restored {
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
-			if err := ep.RLockAlive(); err != nil {
-				ep.LogDisconnectedMutexAction(err, "before filtering labels during regenerating restored endpoint")
+			if err := ep.RegenerateAfterRestore(); err != nil {
 				epRegenerated <- false
 				return
 			}
-			scopedLog := log.WithField(logfields.EndpointID, ep.ID)
-			// Filter the restored labels with the new daemon's filter
-			l, _ := labels.FilterLabels(ep.OpLabels.AllLabels())
-			ep.RUnlock()
-
-			allocateCtx, cancel := context.WithTimeout(context.Background(), option.Config.KVstoreConnectivityTimeout)
-			defer cancel()
-			identity, _, err := cache.AllocateIdentity(allocateCtx, d, l)
-
-			if err != nil {
-				scopedLog.WithError(err).Warn("Unable to restore endpoint")
-				epRegenerated <- false
-				return
-			}
-
-			// Wait for initial identities and ipcache from the
-			// kvstore before doing any policy calculation for
-			// endpoints that don't have a fixed identity or are
-			// not well known.
-			if !identity.IsFixed() && !identity.IsWellKnown() {
-				identityCtx, cancel := context.WithTimeout(context.Background(), option.Config.KVstoreConnectivityTimeout)
-				defer cancel()
-
-				err = cache.WaitForInitialGlobalIdentities(identityCtx)
-				if err != nil {
-					scopedLog.WithError(err).Warn("Failed while waiting for initial global identities")
-					return
```

Comment:
hmm, not doing `epRegenerated <- false` here might have been a bug before! (it's implicitly fixed by this refactor).

Reply:
🤔

Reply:
🤔 - filed an issue to track this for v1.6 - https://github.com/cilium/cilium/issues/9007